File: Examples/VisualBasic/WorkingWithTasks/ReadStopResumeDates.vb
Imports System.IO
Imports Aspose.Tasks.Util

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace WorkingWithTasks
    Public Class ReadStopResumeDates
        Public Shared Sub Run()
            ' ExStart:ReadStopResumeDates
            ' Read project from file stream
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)
            Dim fs As New FileStream(dataDir & "StopResumeDates.mpp", FileMode.Open)
            Dim prj As New Project(fs)
            fs.Close()

            ' Create a ChildTasksCollector instance
            Dim collector As New ChildTasksCollector()

            ' Collect all the tasks from RootTask using TaskUtils
            TaskUtils.Apply(prj.RootTask, collector, 0)

            ' Check Stop and Resume dates for all tasks
            For Each tsk1 As Task In collector.Tasks
                If tsk1.[Get](Tsk.[Stop]).ToShortDateString() = "1/1/2000" Then
                    Console.WriteLine("Stop: NA")
                Else
                    Console.WriteLine("Stop: " & tsk1.[Get](Tsk.[Stop]).ToShortDateString())
                End If

                If tsk1.Get(Tsk.[Resume]).ToShortDateString() = "1/1/2000" Then
                    Console.WriteLine("Resume: NA")
                Else
                    Console.WriteLine("Resume: " & tsk1.[Get](Tsk.[Resume]).ToShortDateString())
                End If
            Next
            ' ExEnd:ReadStopResumeDates
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithTasks/ReadStopResumeDates.cs
/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from https://www.nuget.org/packages/Aspose.Tasks/, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using https://forum.aspose.com/c/tasks
*/

namespace Aspose.Tasks.Examples.CSharp.WorkingWithTasks
{
    using System;
    using System.IO;

    using Aspose.Tasks.Util;

    internal class ReadStopResumeDates
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);
            
            //ExStart:ReadStopResumeDates
            //ExFor: Tsk.Stop
            //ExFor: Tsk.Resume
            //ExSummary: Shows how to read task's Stop/Resume dates.
            // Read the project
            var project = new Project(dataDir + "StopResumeDates.mpp");

            // Create a ChildTasksCollector instance
            var collector = new ChildTasksCollector();

            // Collect all the tasks from RootTask using TaskUtils
            TaskUtils.Apply(project.RootTask, collector, 0);

            // Check Stop and Resume dates for all tasks
            foreach (var task in collector.Tasks)
            {
                if (task.Get(Tsk.Stop).ToShortDateString() == "1/1/2000")
                {
                    Console.WriteLine("Stop: NA");
                }
                else
                {
                    Console.WriteLine("Stop: " + task.Get(Tsk.Stop).ToShortDateString());
                }

                if (task.Get(Tsk.Resume).ToShortDateString() == "1/1/2000")
                {
                    Console.WriteLine("Resume: NA");
                }
                else
                {
                    Console.WriteLine("Resume: " + task.Get(Tsk.Resume).ToShortDateString());
                }
            }
            //ExEnd:ReadStopResumeDates
        }
    }
}
